### ISSUE TITLE
Deploying prisma to CloudFlare pages using Nuxt/Nitro and node-postgres (pg) is using the wrong(vercel) wasm path

### ISSUE THREAD

**Issue by langleyd** [prisma/prisma#23500]
### Bug description

Deploying prisma to CloudFlare pages using Nuxt/Nitro and node-postgres (pg) seems to be using the wrong wasm path as if the current runtime has been detected as vercel.

It also happpens locally when using nitro with the `cloudflare-pages` present

Error:
```
Building Nuxt Nitro server (preset: cloudflare_pages)                                                                          nitro 11:55:06 PM

[nitro 11:55:07 PM]  ERROR  Error: Could not load /Users/.../node_modules/.prisma/client/query_engine_bg.wasm?module (imported by node_modules/.prisma/client/wasm-edge-light-loader.js): ENOENT: no such file or directory, open '/Users/.../node_modules/.prisma/client/query_engine_bg.wasm?module'
```

### How to reproduce

1. Create a nuxt app with prisma
2. Set the cloudflare preset nuxt.config.ts:
``` 
nitro: {
    preset: 'cloudflare-pages'
  },
```
3. run  `npx nuxi build`

### Expected behavior

Cloudflare is not detected as `edge-light` runtime and loads the correct wasm.

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS
- Database: Postgre with adapter-pg

### Prisma Version

```
5.11.0
```

**Comment by SevInf** [comment]
Hi @langleyd 
This is a bug on Nitro's side. There is unfortunately no workaround we can do on our side. We've already reported it [here](https://github.com/unjs/nitro/issues/2242), you can follow the progress on Nitro repo.

There is also another problem in nitro's wasm handling and we are working with them to fix it as well.

**Comment by jkomyno** [comment]
Hi @langleyd, may you please try this new unofficial version of Prisma, and report back with the results on your Nuxt app? `5.15.0-integration-client-dynamic-wasm-imports.1`

You can quickly install it via:

```sh
pnpm add -D prisma@5.15.0-integration-client-dynamic-wasm-imports.1;
pnpm update "@prisma/*" 5.15.0-integration-client-dynamic-wasm-imports.1
```

(please double check that prisma generate has run before retrying).
Thanks!

**Comment by jkomyno** [comment]
> Hi @langleyd, may you please try this new unofficial version of Prisma, and report back with the results on your Nuxt app? `5.15.0-integration-client-dynamic-wasm-imports.1`
> 
> You can quickly install it via:
> 
> ```shell
> pnpm add -D prisma@5.15.0-integration-client-dynamic-wasm-imports.1;
> pnpm update "@prisma/*" 5.15.0-integration-client-dynamic-wasm-imports.1
> ```
> 
> (please double check that prisma generate has run before retrying). Thanks!

Bad news (thanks @SevInf for the internal report): apparently, the application works locally with `pnpm dev`, but it fails once built with
> [10:54:40 AM]  ERROR  Could not load /private/tmp/prisma-driver-adapters-nuxt/node_modules/.pnpm/@prisma+client@5.15.0-integration-client-dynamic-wasm-imports.1_prisma@5.15.0-integration-client-dynamic-wasm-imports.1/node_modules/.prisma/client/query_engine_bg.wasm?module (imported by node_modules/.pnpm/@prisma+client@5.15.0-integration-client-dynamic-wasm-imports.1_prisma@5.15.0-integration-client-dynamic-wasm-imports.1/node_modules/.prisma/client/wasm-edge-light-loader.js): ENOENT: no such file or directory, open '/private/tmp/prisma-driver-adapters-nuxt/node_modules/.pnpm/@prisma+client@5.15.0-integration-client-dynamic-wasm-imports.1_prisma@5.15.0-integration-client-dynamic-wasm-imports.1/node_modules/.prisma/client/query_engine_bg.wasm?module'

**Comment by jkomyno** [comment]
Update: this issue goes away when using the version mentioned [here](https://github.com/prisma/prisma/issues/23500#issuecomment-2135608015) + setting 

```js
nitro: {
  experimental: {
    wasm: true
  }
}
```

in the `nuxt.config.js` configuration file.

**Comment by jkomyno** [comment]
Hi @langleyd, we're closing this issue via https://github.com/prisma/docs/issues/6054 + https://github.com/prisma/team-orm/issues/1164. You can now refer to this documentation: https://www.prisma.io/docs/orm/overview/databases/database-drivers#nuxt.

We also recommend using Prisma 5.15.0 on Nuxt @ Edge. Thanks!

**Comment by themathmagician** [comment]
I am developing a small open source Nuxt application, targeted for Cloudflare and D1.
I was following the [Prisma for D1 guide](https://www.prisma.io/docs/guides/using-prisma-orm-with-cloudflare-d1), when I ran into trouble - the app could not deploy anymore to Cloudflare.
```
ERROR  Error: Could not load /Users/agatanoair/workspace/theslope/node_modules/.prisma/client/query_engine_bg.wasm?module (imported by node_modules/.prisma/client/wasm-edge-light-loader.mjs): ENOENT: no such file or directory, open '/Users/agatanoair/workspace/theslope/node_modules/.prisma/client/query_engine_bg.wasm?module'
```
The issue still appears with   "nuxt": "^3.15.0", and     "prisma": "^6.2.1", the app runs locally, and can generate tables both locally and remote, but fails when you try to deploy to Cloudflare. Here is a link to the [failing Github Actions](https://www.prisma.io/docs/guides/using-prisma-orm-with-cloudflare-d1)
I solved it after reading through this thread, by adding to nuxt.config.ts
```
  nitro: {
    preset: "cloudflare_module",
    experimental: {
      wasm: true
    }
  },
```
It was maybe worth adding information about this workaround to the documentation?
